### PR TITLE
Pass along the exception code.

### PR DESCRIPTION
**Pull request by SpacePossum**
Quick fix for (not yet released) configuration exception construction
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1586/files#r48208398

### CODE REVIEW

File: Symfony/CS/ConfigurationException/InvalidConfigurationException.php [hunk 1]
```diff
@@ -24,9 +24,10 @@ class InvalidConfigurationException extends \InvalidArgumentException
 {
     /**
      * @param string $message
+     * @param int    $code
      */
-    public function __construct($message)
+    public function __construct($message, $code = FixCommand::EXIT_STATUS_FLAG_HAS_INVALID_CONFIG)
```
Comment (GrahamCampbell) [inline Symfony/CS/ConfigurationException/InvalidConfigurationException.php]: Maybe make the the default `null`, then do a ternary thing inside the code. That would be cleaner to me.

Comment (GrahamCampbell) [inline Symfony/CS/ConfigurationException/InvalidConfigurationException.php]: Something like this:

```php
    /**
     * @param string   $message
     * @param int|null $code
     */
    public function __construct($message, $code = null)
    {
        parent::__construct($message, $code ?: FixCommand::EXIT_STATUS_FLAG_HAS_INVALID_CONFIG);
    }
```

Comment (keradus) [inline Symfony/CS/ConfigurationException/InvalidConfigurationException.php]: :+1:

Comment (GrahamCampbell) [inline Symfony/CS/ConfigurationException/InvalidConfigurationException.php]: Just what I'd do. D'no what anyone else thinks?

@keradus?

Comment (GrahamCampbell) [inline Symfony/CS/ConfigurationException/InvalidConfigurationException.php]: Just missed your :+1: there. ;)

Comment (keradus) [inline Symfony/CS/ConfigurationException/InvalidConfigurationException.php]: https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1630#discussion_r48246011 ;P

Comment (keradus) [inline Symfony/CS/ConfigurationException/InvalidConfigurationException.php]: yeah, 2 s difference :P

Comment (SpacePossum) [inline Symfony/CS/ConfigurationException/InvalidConfigurationException.php]: that would not only check against `null` right? (like `0` would become the default as well?)

Comment (keradus) [inline Symfony/CS/ConfigurationException/InvalidConfigurationException.php]: feel free to use strict comparision

Comment (SpacePossum) [inline Symfony/CS/ConfigurationException/InvalidConfigurationException.php]: :+1: done

File: Symfony/CS/ConfigurationException/InvalidConfigurationException.php [hunk 2]
```diff
@@ -24,9 +24,10 @@ class InvalidConfigurationException extends \InvalidArgumentException
 {
     /**
      * @param string $message
+     * @param int    $code
```
Comment (GrahamCampbell) [inline Symfony/CS/ConfigurationException/InvalidConfigurationException.php]: int|null

Comment (SpacePossum) [inline Symfony/CS/ConfigurationException/InvalidConfigurationException.php]: another down side of hiding the default form the signature

Comment (keradus) [inline Symfony/CS/ConfigurationException/InvalidConfigurationException.php]: It is general style of default values - which in this one case indeed gives you nothing but consistency.

General issue:
`function foo($a, $b = 1, $c = 2) {}`
vs
`function foo($a, $b = null, $c = null) {}`

Try to pass $a and $c, but leave the $b as default in first example. You must know the default, and when it changes you need to change calls. In second example you just pass null. Everything would be cool if `default` keyword would be introduced, but it is not.
Also, that strategy of dealing is also used by native PHP functions.

Comment (GrahamCampbell) [inline Symfony/CS/ConfigurationException/InvalidConfigurationException.php]: > another down side of hiding the default form the signature

I don't agree that it's a downside at all. Infact, I see it as better that the alternative.

Comment (GrahamCampbell) [inline Symfony/CS/ConfigurationException/InvalidConfigurationException.php]: It allows you to skip defaults by passing null.

Comment (SpacePossum) [inline Symfony/CS/ConfigurationException/InvalidConfigurationException.php]: agree to disagree, update :)

**Comment by keradus** [comment]
:+1:

**Comment by GrahamCampbell** [comment]
:+1:

**Comment by gharlan** [comment]
:+1:

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request merged**